debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9MemTagHelper.java

```diff
@@ -266,8 +265,9 @@ public static VoidPointer j9mem_get_footer_padding(J9MemTagPointer headerEyeCatc
 	
 	public static UDATA ROUNDED_FOOTER_OFFSET(UDATA number) 
 	{
-		UDATA a = (number.add(ROUNDING_GRANULARITY - 1)).add(J9MemTag.SIZEOF);
-		UDATA b = new UDATA(ROUNDING_GRANULARITY - 1).bitNot();
+		UDATA mask = new UDATA(ROUNDING_GRANULARITY - 1);
+		UDATA a = mask.add(number).add(J9MemTag.SIZEOF);
+		UDATA b = mask.bitNot();
```

Comment:
Based on the PR description
> This change modifies ROUNDED_FOOTER_OFFSET() so it always returns a UDATA object independent of the actual
argument type, thus avoiding that exception for blocks that otherwise would fail the sanity checks.

I'm not clear on why the original code wouldn't have already always returned a UDATA.  Can you explain why this change is required?

Reply:
U64.add() returns a U64 while UDATA.add(UScalar) returns UDATA; UDATA.longValue() doesn't throw an exception.

Reply:
Which variable is a U64 here? `J9MemTag.SIZEOF`?

Reply:
Thanks Keith.  I had forgotten that `U64` was changed to be a subclass of `UDATA`